### ISSUE TITLE
Spike: Investigate and apply accessibility reqs. to the dataset page

### ISSUE THREAD

**Issue by TaniaSchlatter** [IQSS/dataverse#6072]
This spike is to learn about and practice applying accessibility best practices to an existing page.
Outcomes:

- Identification of standards that are most relevant to Dataverse design and development (see draft below)
- Identification of tools and resources we can use to help us improve accessibility (some included below)
- Identification of changes to design and development processes that will help us improve accessibility
- More accessible code for the dataverse page
- Better position for working with automatic testing processes 

Informational resources:
1. [Conceptual overview of WCAG](https://docs.google.com/document/d/1NrJjiRMHvs5U0XOPeBaYDoQ6lae_9-SO8j5ezF_vM0k/edit)
For the purposes of [Harvard’s Digital Accessibility Policy](https://accessibility.huit.harvard.edu/digital-accessibility-policy), the University uses the Worldwide Web Consortium’s Web Content Accessibility Guidelines version 2.1, Level AA.
2. Easy checks and methods to address: ["Easy Checks - A First Review of Web Accessibility" ](https://www.w3.org/WAI/test-evaluate/preliminary/)recommended by the Harvard Library usability group.
3. Standards in detail, methods, and measurements: [How to meet WCAG 2.1](https://www.w3.org/WAI/WCAG21/quickref/) from W3C WCAG group.
-----
Links to detailed guidelines on the W3C site that are most relevant to the Dataverse application (not the content users add/upload):

[Guideline 1.1 – Text Alternatives](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=111#text-alternatives)
Provide text alternatives for any non-text content so that it can be changed into other forms people need, such as large print, braille, speech, symbols or simpler language.

[Guideline 1.3 – Adaptable](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=111#adaptable)
Create content that can be presented in different ways (for example simpler layout) without losing information or structure.

[Guideline 1.4 – Distinguishable](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=111#distinguishable)
Make it easier for users to see and hear content including separating foreground from background.

Test color contrast with this tool: https://github.com/ThePacielloGroup/CCA-OSX/releases/tag/2.4

[Guideline 2.1 – Keyboard Accessible](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=111#keyboard-accessible)
Make all functionality available from a keyboard.
[How to test](https://accessibility.digital.gov/visual-design/keyboard-access/) | Harvard-only [slides of how to check and ARIA tags](https://drive.google.com/file/d/1q6N1OojYysZJIWm-n70BHR6ffr977d0e/view)

[Guideline 2.4 – Navigable](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=111#navigable)
Provide ways to help users navigate, find content, and determine where they are.

[Guideline 2.5 – Input Modalities](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=111#input-modalities)
Make it easier for users to operate functionality through various inputs beyond keyboard.

[Guideline 3.2 – Predictable](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=111%2C312#predictable)
Make Web pages appear and operate in predictable ways.

[Guideline 3.3 – Input Assistance](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=111%2C312#input-assistance)
Help users avoid and correct mistakes.

[Guideline 4.1 – Compatible](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=111%2C312%2C413#compatible)
Maximize compatibility with current and future user agents, including assistive technologies.

**Comment by TaniaSchlatter** [comment]
We'll review initial status in the design meeting Sept. 3.

**Comment by mheppler** [comment]
Inventory of accessibility issues created as a result of exploring these new tools.

- Images - Add/improve alt text, link title, components #6106
- Metadata Labels - Review use of `label` tag outside of form (viewMode) #6109
- Code Cleanup - `i` tag used for icons bad practice #6144

**Comment by mheppler** [comment]
Comments from @kcondon and @TaniaSchlatter were compiled into this outline.

PROGRESS REPORT
++++++++++++++++

1. standard Harvard uses for accessibility (https://accessibility.huit.harvard.edu/policies)
2. plan for compliance, including date (December 1, 2019)
3. tools we use as part of a specific accessibility check (e.g. AMP, Siteimprove/Chrome extension, ++color contrast++; https://accessibility.huit.harvard.edu/tools)
4. strategy for addressing current and future accessibility issues

TO-DO
++++++

- [ ] Siteimprove report
- [ ] AMP report
- [ ] VoiceOver test
- [ ] specific dataset pg issues from Siteimprove + AMP reports
- [ ] describe process/knowledge/tools used to evaluate accessibility
- [ ] list issues found on dataset page (e.g. "X separate issues that need fixing, see screen shot")

FOLLOW UP
++++++++++

1. group/individual accessibility issues
2. discuss with team/Danny how to incorporate tools into dev/QA process
3. evaluate fix current dataset issues vs factor into upcoming redesign
4. share accessibility tips/resources/expectations with community members

**Comment by djbrooke** [comment]
@mheppler @TaniaSchlatter @jggautier thanks for the post-standup discussion about this. 

@mheppler, the plan we discussed made sense to me, please feel free to summarize here and close. I'll be happy to work with you in #6106 and other issues to communicate the knowledge sharing

**Comment by mheppler** [comment]
I finally managed to squeeze out a manual report of the dataset pg in AMP, which checked off the to-do above. That left outlining the "process", and to deliver on that, I have added a new "Accessibility Testing" section to the Testing pg in the Developer Guide (see PR #6218).

The impact of accessibility on our development process will be evolving as we learn more about the tools, and requirements. The image alt text issue #6106, will provide out team a "proof of process" of sorts.

Reports from Siteimprove and AMP both caught this issue on the dataset pg, and when developing a fix, the browser tools from Siteimprove, AMP and Wave will all be checked, to make sure a suitable solution is implemented. Then once a pull request is submitted and approved, a test plan will be reviewed with @kcondon and @djbrooke to make sure QA knows not only what to test, but how to test.
